test/simplegraphs/generators/randgraphs.jl

```diff
@@ -15,6 +15,21 @@
         @test eltype(DiGraph{T}(5, 2)) == T
     end
 
+    pdg = PathDiGraph(5)
+    for dg in testdigraphs(pdg)
+        g = Graph(dg)
+        dg10 = copy(dg); add_vertices!(dg10, 5)
+        add_vertices!(g, 5)
+        d1 = @inferred(DiGraph(g, 0.0, 1.0))
+        @test d1 == reverse(dg10)
+        d2 = @inferred(DiGraph(g, 2//7, 5//9))
+        @test ne(d2) >= ne(g)
+        d3 = @inferred(DiGraph(g, 0, 0))
+        @test d3 == DiGraph(g)
+        d4 = @inferred(DiGraph(g, -1, 200))
+        @test d4 == reverse(dg10)
+    end
+
```

Comment:
Maybe add a test, that verifies, that we get the same result when using the same seed.
Also some tests for edge cases, such as the zero graph, graphs without edges and graphs with self-loops.

Reply:
I'm good with all of that except for the self-loops. We're not really supporting them (they're definitely second-class).